R/theme-elements.r

```diff
@@ -244,144 +244,277 @@ element_grob.element_line <- function(element, x = 0:1, y = 0:1,
   )
 }
 
-
-
-# Define an element's class and what other elements it inherits from
-#
-# @param class The name of class (like "element_line", "element_text",
-#  or the reserved "character", which means a character vector (not
-#  "character" class)
-# @param inherit A vector of strings, naming the elements that this
-#  element inherits from.
+#' Define an element's class and what other elements it inherits from
+#'
+#' @param class The name of class (like "element_line", "element_text",
+#'  or the reserved "character", which means a character vector (not
+#'  "character" class)
+#' @param inherit A vector of strings, naming the elements that this
+#'  element inherits from.
+#' @export
 el_def <- function(class = NULL, inherit = NULL, description = NULL) {
-  list(class = class, inherit = inherit, description = description)
+  structure(
+    list(class = class, inherit = inherit, description = description),
+    class=c('el_def','list')
```

Comment:
This is makes it possible to enforce the expected arguments when using the add_element(...) function in the R6 class below.